src/main/java/org/jitsi/videobridge/simulcast/SimulcastReceiver.java

```diff
@@ -188,32 +189,76 @@ public void setSimulcastStreams(long[] ssrcs)
     /**
      * Sets the simulcast streams for this receiver and fires an event about it.
      *
-     * @param simulcastStreams the simulcast streams for this receiver.
+     * @param newSimulcastStreams the simulcast streams for this receiver.
      */
-    public void setSimulcastStreams(SimulcastStream[] simulcastStreams)
+    public void setSimulcastStreams(SimulcastStream[] newSimulcastStreams)
     {
-        this.simulcastStreams = simulcastStreams;
+        SimulcastStream[] oldSimulcastStreams = this.simulcastStreams;
+        int oldLen
+            = oldSimulcastStreams == null ? 0 : oldSimulcastStreams.length;
+        int newLen
+            = newSimulcastStreams == null ? 0 : newSimulcastStreams.length;
+
+        if (oldLen == newLen)
```

Comment:
Can we simplify this? Maybe use something like:
boolean equal = (oldLen == 0 && newLen == 0) || Arrays.equals(oldSimulcastStreams, newSimulcastStreams);